### ISSUE TITLE
restrict-full-import fails on "empty" array pattern

### ISSUE THREAD

**Issue by lemonmade** [Shopify/eslint-plugin-shopify#222]
Failing code:

```js
const [, foo] = bar;
```

we try to access `type`, but the empty slot is actually just `null`:

<img width="1278" alt="screen shot 2019-02-15 at 11 53 56 am" src="https://user-images.githubusercontent.com/3012583/52871657-890fc900-3118-11e9-8f8e-6f72a1468055.png">
